### ISSUE TITLE
Sort by Date/Name

### ISSUE THREAD

**Issue by SparkyTemper** [linuxmint/xviewer#22]
Would it be possible to add a sort feature?  With large folders, it would be nice to see the newest file.

**Comment by mitap94** [comment]
Other viewer apps follow the sort order as is set in the folder, it would definitely be better if this one does also.

**Comment by jversusk** [comment]
### Use file manager sort order

It makes sense for apps like Pix and Gwenview to have their own sorting options, because they are both viewer and browser. But Xviewer is only a viewer, so it should just use whatever sort order the file manager has for the source folder.

I haven't been able to find any Linux image viewer that does this - it's the single biggest issue I've faced ever since I switched from Windows years ago. While it might be impractical to integrate every image viewer with every file manager in this way, **it'd be nice if at least Xviewer & Nemo on Mint had this integration**. If necessary, it can be implemented as an extension/patch that interested users can install separately.

**Comment by stephan-dev** [comment]
I have no idea how to link XViewer file sorting, with Nemo file sorting.
I can barely read the C language.
But I think I found where in the code the sorting is defined : [xviewer-list-store.c](https://github.com/linuxmint/xviewer/blob/a54362c01a796dd77e0bfabd6ed608813b03ea98/src/xviewer-list-store.c)

line 173, line 553, line 619
 `gtk_tree_sortable_set_sort_column_id`

reference [Gnome Tree Sortable](https://developer.gnome.org/gtk3/stable/GtkTreeSortable.html#gtk-tree-sortable-set-sort-column-id)
[gtk+ by example, tree view, sorting](https://en.wikibooks.org/wiki/GTK%2B_By_Example/Tree_View/Sorting)

I don't know where the "default" column for sorting comes from, if it's set by xviewer (I couldn't find it), or a global variable (idk where to look).

I'm not sure if that code defines the sorting order when the user presses the right arrow key (next image) or if it is for sorting the internal gallery view, or both.

My 2 cents, maybe someone can take the work from here.

**Comment by AndreiMiculita** [comment]
> ### Use file manager sort order
> 
> It makes sense for apps like Pix and Gwenview to have their own sorting options, because they are both viewer and browser. But Xviewer is only a viewer, so it should just use whatever sort order the file manager has for the source folder.
> 
> I haven't been able to find any Linux image viewer that does this - it's the single biggest issue I've faced ever since I switched from Windows years ago. While it might be impractical to integrate every image viewer with every file manager in this way, **it'd be nice if at least Xviewer & Nemo on Mint had this integration**. If necessary, it can be implemented as an extension/patch that interested users can install separately.

I also agree, I often find myself needing this. I think this could be done through the plugin system.

**Comment by vegam57** [comment]
I'm a common user who can't write code, but I'm looking forward to implementing this feature.

**Comment by thegame1999** [comment]
This can be closed as nemo does this by default via the "image-viewers-with-external-sort" option.
